app/controllers/concerns/curation_concerns/file_sets_controller_behavior.rb

```diff
@@ -71,6 +71,14 @@ def show
 
     def destroy
       parent = @file_set.in_works.first
+      # When running the demo application this line is needed to
+      # make sure the work does not point to the file_set that
+      # is being deleted. HOWEVER, the RSpec test added in this PR
+      # works even without it.
+      #
+      # Any ideas?
+      #
+      unlink_from_work(@file_set, parent)
```

Comment:
@jcoyne @mjgiarlo: Any ideas why does the test **pass** without the fix but the internal application blows up without it??